components/content/ProseCode.vue

```diff
@@ -0,0 +1,68 @@
+<script setup>
+import mermaid from 'mermaid';
+
+const props = defineProps({
+  code: {
+    type: String,
+    default: ''
+  },
+  language: {
+    type: String,
+    default: ''
+  },
+  filename: {
+    type: String,
+    default: null
+  },
+  highlights: {
+    type: Array,
+    default: () => []
+  }
+});
+
+onMounted(() => {
```

Comment:
Will this hook be called for static site generation?

Reply:
I saw that in the Vue docs that this hook not called when using server side rendering.
Are our docs a static site? If so, this hook will be called. 

https://nuxtjs.org/docs/features/rendering-modes#server-side-rendered-sites-and-static-sites

I have looked at the static site at port `:5001` and I can see the diagram as expected.

Reply:
Sweet. I'm not really sure if ours is ssr or static to be honest. We don't have `ssr: false` or [target: static](https://nuxtjs.org/docs/features/deployment-targets/) anywhere in our codebase but we still use `yarn generate` which outputs a bunch of static files

![image](https://user-images.githubusercontent.com/70119888/208766314-b30823f6-fa72-427d-9698-8af9d3218211.png)

Reply:
I remember our initial plan was to go fully static but we were using an unstable release of nuxt and so `target: static` was buggy.

Maybe this is something we should get sorted now that its a stable release? - before releasing lots of code that may or may not work with our preferred rendering method

Reply:
@AndyClifford we have now added `target: static` and have tested it, seems to work as expected.

Reply:
@AndyClifford Tom and I had a look at this and it appears as though our current config is correct.
The docs you linked are for Nuxt v2 🧡

Reply:
Sweet - okay looks like nuxt.com is the go to as opposed to nuxtjs.org

Reply:
Yeah it's a little confusing 😵